src/create-glamorous.js

```diff
@@ -149,6 +149,13 @@ function createGlamorous(splitProps) {
         },
       })
 
+      function withComponent(newComp, options) {
+        return glamorous(newComp, {
+          forwardProps: GlamorousComponent.forwardProps,
+          ...options,
```

Comment:
For some reason I always thought this would throw an exception when `undefined`, but I see by your unit tests, that's not the case!